File: Examples/IndoorLocationWithARKit/IndoorLocationWithARKit/ViewController.m
#import "ViewController.h"
#import <EstimoteSDK/EstimoteSDK.h>
#import "EILIndoorSDK.h"

@interface ViewController () <EILIndoorLocationManagerDelegate>

@property (nonatomic, strong) EILIndoorLocationManager *indoorLocationManager;
@property (nonatomic, strong) EILLocation *location;

@property (nonatomic, strong) EILIndoorLocationView *locationView;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    // TODO: put your App ID and App Token here
    // You can get them by adding your app on https://cloud.estimote.com/#/apps
    [ESTConfig setupAppID:@"<#App ID#>" andAppToken:@"<#App Token#>"];
    
    self.indoorLocationManager = [EILIndoorLocationManager new];
    self.indoorLocationManager.delegate = self;
    self.indoorLocationManager.mode = EILIndoorLocationManagerModeExperimentalWithARKit;
    [self.indoorLocationManager start];
    
    EILRequestFetchLocation *request = [[EILRequestFetchLocation alloc] initWithLocationIdentifier:@"<#my-location#>"];
    [request sendRequestWithCompletion:^(EILLocation * _Nullable location, NSError * _Nullable error) {
        if (location)
        {
            self.location = location;
            
            self.locationView = [[EILIndoorLocationView alloc] initWithFrame:self.view.frame];
            self.locationView.rotateOnPositionUpdate = NO;
            [self.locationView drawLocation:self.location];
            [self.view addSubview:self.locationView];
            
            [self.indoorLocationManager startPositionUpdatesForLocation:self.location];
            
        } else {
            NSLog(@"Can't fetch location: %@", error);
        }
    }];
}

- (void)indoorLocationManager:(EILIndoorLocationManager *)manager
            didUpdatePosition:(EILOrientedPoint *)position
                 withAccuracy:(EILPositionAccuracy)positionAccuracy
                   inLocation:(EILLocation *)location {
    EILPoint *positionInARKitCoordinates = [self.indoorLocationManager ARKitPointFromIndoorLocationPoint:position];
    NSLog(@"IL Position x: %5.2f, y: %5.2f, orientation: %3.0f", position.x, position.y, position.orientation);
    NSLog(@"ARKit Position x: %5.2f, y: %5.2f", positionInARKitCoordinates.x, positionInARKitCoordinates.y);
    [self.locationView updatePosition:position];
}

- (void)indoorLocationManager:(EILIndoorLocationManager *)manager didFailToUpdatePositionWithError:(NSError *)error {
    NSLog(@"Failed to update position: %@", error);
}

@end

File: Examples/IndoorMap/IndoorMap/ViewController.swift
//
//  ViewController.swift
//  IndoorMap
//

import UIKit

class ViewController: UIViewController, EILIndoorLocationManagerDelegate {

    var locationManager: EILIndoorLocationManager!

    var location: EILLocation!

    @IBOutlet weak var locationView: EILIndoorLocationView!

    override func viewDidLoad() {
        super.viewDidLoad()

        // TODO: put your App ID and App Token here
        // You can get them by adding your app on https://cloud.estimote.com/#/apps
        ESTConfig.setupAppID("<#App ID#>", andAppToken: "<#App Token#>")

        self.locationManager = EILIndoorLocationManager()
        self.locationManager.delegate = self
        self.locationManager.mode = EILIndoorLocationManagerMode.standard

        // TODO: replace with an identifier of your own location
        // You will find the identifier on https://cloud.estimote.com/#/locations
        let fetchLocationRequest = EILRequestFetchLocation(locationIdentifier: "<#my-location#>")
        fetchLocationRequest.sendRequest { (location, error) in
            if let location = location {
                self.location = location

                // You can configure the location view to your liking:
                self.locationView.showTrace = true
                self.locationView.rotateOnPositionUpdate = false
                // Consult the full list of properties on:
                // http://estimote.github.io/iOS-Indoor-SDK/Classes/EILIndoorLocationView.html

                self.locationView.drawLocation(location)
                self.locationManager.startPositionUpdates(for: self.location)
            } else if let error = error {
                print("can't fetch location: \(error)")
            }
        }
    }

    func indoorLocationManager(_ manager: EILIndoorLocationManager, didFailToUpdatePositionWithError error: Error) {
        print("failed to update position: \(error)")
    }

    func indoorLocationManager(_ manager: EILIndoorLocationManager, didUpdatePosition position: EILOrientedPoint, with positionAccuracy: EILPositionAccuracy, in location: EILLocation) {
        print(String(format: "x: %5.2f, y: %5.2f, orientation: %3.0f", position.x, position.y, position.orientation))

        self.locationView.updatePosition(position)
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

}
